### PR TITLE
7125 export import api

### LINKED ISSUE #7125: Export/import API
Export/import feature could be used through Import API before than having UI integration (#7056).

- [X] Exports should run in Resque.
- [X] Logging.
- [X] Working without S3.
- [X] API should allow choosing datasets by user table id. Only layers with all related tables available (selected & readable) will be exported.

cc @javisantana @xavijam @gfiorav

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #7125.

### CODE REVIEW

**Comment by juanignaciosl** [comment]
@javitonino could you CR this, please? @gfiorav has made a lot of progress with the UI so this should be close to the final code.

File: app/controllers/carto/api/visualization_exports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      def show
+        render_jsonp(VisualizationExportPresenter.new(@visualization_export).to_poro, 200)
+      end
+
+      private
+
+      def load_visualization
+        visualization_id = params[:visualization_id]
+        @visualization = load_visualization_from_id_or_name!(visualization_id)
```
Comment (javitonino) [inline app/controllers/carto/api/visualization_exports_controller.rb]: I think it's better if we enforce the parameter to be `visualization.id` here (as in UUID). As this is a new controller, I think we should avoid the mess that is `load_visualization_from_id_or_name`. Guido confirmed he always sends the UUID.

File: app/controllers/carto/api/visualization_exports_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      end
+
+      def load_visualization_export
+        id = params[:id]
```
Comment (javitonino) [inline app/controllers/carto/api/visualization_exports_controller.rb]: Better to use `id = uuid_parameter(:id)` from `ControllerHelper`. This ensures the UUID is correct, as postgres complains if the format is not right.

File: app/models/carto/visualization_export.rb [hunk 3]
```diff
@@ -1,107 +1,82 @@
...
+
+      logger.append('Deleting tmp file')
+      FileUtils.rm(filepath)
+
+      logger.append('Finishing')
+      state = filepath.present? && url.present? ? STATE_COMPLETE : STATE_FAILURE
+      update_attributes(state: state, file: filepath, url: url)
+      true
+    rescue => e
+      logger.append_exception('Exporting', exception: e)
```
Comment (javitonino) [inline app/models/carto/visualization_export.rb]: Maybe you should set the state to failure here (if it is not being set in the caller).

File: script/resque [hunk 4]
```diff
@@ -1,2 +1,2 @@
 #!/bin/sh
-VVERBOSE=true QUEUE=imports,users,geocodings,synchronizations,tracker rake environment resque:work
+VVERBOSE=true QUEUE=imports,exports,users,geocodings,synchronizations,tracker rake environment resque:work
```
Comment (javitonino) [inline script/resque]: Please, test this in the on-premise container. I think they use a different start-up script for resque there.

Comment (juanignaciosl) [inline script/resque]: Ok, I will. In fact I'm pinging @azamorano so he can look for needed changes. I also have to update NEWS yet.

**Comment by javitonino** [comment]
I would like to see a change so the controller only accepts UUID parameters. After that (and checking on-premise) :+1:

File: app/controllers/carto/api/visualization_exports_controller.rb (hunk 1 above)
Comment (juanignaciosl) [inline app/controllers/carto/api/visualization_exports_controller.rb]: Ok.

File: script/resque (hunk 4 above)
Comment (azamorano) [inline script/resque]: We do indeed but is the one inherited from staging and production environments, so once the modifications are done there they should be added to on-premise too.

File: app/controllers/carto/api/visualization_exports_controller.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/controllers/carto/api/visualization_exports_controller.rb]: Ok.

File: app/models/carto/visualization_export.rb (hunk 3 above)
Comment (juanignaciosl) [inline app/models/carto/visualization_export.rb]: All state handling is done in this method, so rescue should certainly have it 👍  . I'll add it after the logging. Thanks!

**Comment by juanignaciosl** [comment]
@javitonino since there've been some commits take a last look, please. All changes have been minor anyway.

**Comment by javitonino** [comment]
:ship: it! Let's demo this thing!

**Pull request merged**